Ryujinx/Ui/ApplicationLibrary.cs

```
@@ -52,26 +52,31 @@ public static void LoadApplications(List<string> appDirs, VirtualFileSystem virt
+                    if (!Directory.Exists(appDir))
+                    {
+                        Logger.PrintWarning(LogClass.Application, $"The \"game_dirs\" section in \"Config.json\" contains an invalid directory: \"{appDir}\"");
 
-                    continue;
-                }
+                        continue;
+                    }
 
-                foreach (string app in Directory.GetFiles(appDir, "*.*", SearchOption.AllDirectories))
-                {
-                    if ((Path.GetExtension(app).ToLower() == ".nsp") ||
-                        (Path.GetExtension(app).ToLower() == ".pfs0")||
-                        (Path.GetExtension(app).ToLower() == ".xci") ||
-                        (Path.GetExtension(app).ToLower() == ".nca") ||
-                        (Path.GetExtension(app).ToLower() == ".nro") ||
-                        (Path.GetExtension(app).ToLower() == ".nso"))
+                    foreach (string app in Directory.GetFiles(appDir, "*.*", SearchOption.AllDirectories))
                     {
-                        applications.Add(app);
-                        numApplicationsFound++;
+                        if ((Path.GetExtension(app).ToLower() == ".nsp") ||
+                            (Path.GetExtension(app).ToLower() == ".pfs0") ||
+                            (Path.GetExtension(app).ToLower() == ".xci") ||
+                            (Path.GetExtension(app).ToLower() == ".nca") ||
+                            (Path.GetExtension(app).ToLower() == ".nro") ||
+                            (Path.GetExtension(app).ToLower() == ".nso"))
+                        {
+
+                            applications.Add(app);
+                            numApplicationsFound++;
+
+                        }
                     }
-                }
+                } catch (UnauthorizedAccessException) { }
```

Don't just silently catch the error, log something to say that a path is inaccessible and log the path

---

can you please tell me what should I do with this code? I've got the issue with Ryujinx, and I don't know what to do

---

The catch and braces should also be on new lines

---

At least it don't really fix the issue. We should list all files recursively and test the accessibilty of them.
Right now (after the Xpl0itR nits fixed) it will just throw a message and don't list other files just because one system folder reach the exception.

---

I will do this now @Xpl0itR 

@AcK77 Does it not? It showed the NSP from the root of my drive on my machine.

---

Nevermind, I don't know why the NSP shows I can see this code shouldn't work, sorry. Will fix.